File: WebContent/cqut/hrms/view/employee/js/EmployeeService.js
var employeeApp = angular.module("employeeApp", []);

employeeApp.factory('EmployeeService',['$resource',function($resource){
	
	var employeeService = $resource('/Employee/:action', {});
	
	employeeService.insertEmployee=function(employee,sucesscb,errorcb){
		
		employeeService.save({action:"insertEmployee"},employee,sucesscb,errorcb);
	}
	
	employeeService.getEmployeeByPermissionId=function(permissionId,sucesscb,errorcb){
		
		employeeService.save({action:"getEmployeeByPermissionId"},permissionId,sucesscb,errorcb);
	}
	
	employeeService.getEmployee=function(sucesscb,errorcb){
		
		employeeService.query({action:"getEmployee"},sucesscb,errorcb);
	}
	
	employeeService.getEmployeeByDepartmentId=function(departmentId,sucesscb,errorcb){
		
		employeeService.save({action:"getEmployeeByDepartmentId"},departmentId,sucesscb,errorcb);
	}
	
	employeeService.getEmployeeByName=function(name,sucesscb,errorcb){
		
		employeeService.save({action:"getEmployeeByName"},name,sucesscb,errorcb);
	}
	
	employeeService.getEmployeeByDepartmentIdAndName=function(departmentId,name,sucesscb,errorcb){
		
		employeeService.save({action:"getEmployeeByDepartmentIdAndName"},{departmentId:departmentId,name:name},sucesscb,errorcb);
	}
	
	return employeeService;
	
}]);

File: src/com/cqut/service/employee/EmployeeService.java
package com.cqut.service.employee;

import java.util.List;

import org.apache.ibatis.annotations.Param;

import com.cqut.entity.Employee;

public interface EmployeeService {
	 
	 public int insertEmployee(Employee employee);
	 
	 public List<Employee> getEmployee();
	 
	 public List<Employee> getEmployeeByPermissionId(int permissionId);
	 
	 public List<Employee> getEmployeeByDepartmentId(int departmentId);
		
	 public List<Employee> getEmployeeByName(String name);
		
	 public List<Employee> getEmployeeByDepartmentIdAndName(int departmentId,String name);
}
